prql-compiler/src/parser/chumsky.rs

```diff
@@ -78,3 +85,35 @@ pub fn ident() -> impl Parser<char, Ident, Error = Simple<char>> {
         .chain(just('.').ignore_then(ident_part().or(star)).repeated())
         .map(Ident::from_path::<String>)
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    use insta::assert_yaml_snapshot;
+    #[test]
+    fn test_date() {
+        let date = just('@').ignore_then(
+            digits(4)
+                .then_ignore(just('-'))
+                .then(digits(2))
+                .then_ignore(just('-'))
+                .then(digits(2)),
+        );
+
+        // TODO:
+        // - Why is the result nested?
+        // - What's the type signature to extract `date` to a function?
+        // - Separately from this, but we could be much more liberal about
+        //   dates, this has been discussed on Twitter as a helpful feature.
+        //   (and then we could lint to a standard format, so we're coaching
+        //   users into simplicity)
+        assert_yaml_snapshot!(date.parse("@1984-01-01").unwrap(), @r###"
+        ---
+        - - "1984"
+          - "01"
+        - "01"
```

Comment:
Does anyone know how we could get a sequence of `("1984", "01", "01")`?

Currently I can either get it nested with `.then` (i.e. now), or a vec of char with `.chain`, but I'm guessing I'm doing something wrong? Maybe it's collecting at the wrong place?

Reply:
I think of it this way:
- then combines `before: B` and `after: A` into tuples of `(before, after): (B, A)`,
- chain combines `before: T` and `after: Vec<T>` into vec `result: Vec<T>`.

If you have `just(a).then(b).then(c).then(d)`, the result will be `(((a, b), c), d)`.

If you have `just(a).then(just(b).then(just(c).then(d)))`, you get `(a, (b, (c, d))`

Reply:
I hope this is at least a little helpful.

Reply:
Yes great. 

I had thought there might be a general solution, i.e. without the `.map(|((y, m), d)| (y, m, d));`. Something like `sequence(vec![digits(2), digits(4), digits(2)])` (ignoring the `-` for a second). 

But totally fine without one obv. Thanks

Reply:
...because if we get years, months, days... all the way to microseconds, then that `map` is going to be quite the closure...

Reply:
Yes, if the whole thing is a string, then `.chain` seemed to work well. This approach too.

And at the moment we do just store it as a string in the AST.

WDYT about moving the date parsing to this? I can do that soon-ish; then we'll have our first bit of logic in chumsky.

Reply:
Right — make the pest definition less strict (maybe almost `datetime = { "@" ~ [any] }`, and then pass it to Chumsky to parse?